trade-finance-manager-ui/server/controllers/case/amendments/amendmentFacilityEndDate.get.test.js

```diff
@@ -0,0 +1,269 @@
+          facilityId,
+        },
+        session,
+      };
+
+      await getAmendmentFacilityEndDate(req, res);
+
+      expect(res.redirect).toHaveBeenCalledWith(`/case/${dealId}/facility/${facilityId}/amendment/${amendmentId}/is-using-facility-end-date`);
+    });
+
+    it("should redirect to the 'Has the bank provided a facility end date' page if not using facility end date", async () => {
+      api.getAmendmentById.mockResolvedValueOnce({ status: 200, data: { ...MOCK_AMENDMENT_COVERENDDATE_CHANGE, isUsingFacilityEndDate: false } });
+      api.getFacility = jest.fn().mockResolvedValueOnce(gefFacility);
+
+      const req = {
+        params: {
+          _id: dealId,
+          amendmentId,
+          facilityId,
+        },
+        session,
+      };
+
+      await getAmendmentFacilityEndDate(req, res);
+
+      expect(res.redirect).toHaveBeenCalledWith(`/case/${dealId}/facility/${facilityId}/amendment/${amendmentId}/is-using-facility-end-date`);
+    });
+
+    it('should render the template with isEditable true when the amendment is found and is in progress', async () => {
+      api.getAmendmentById.mockResolvedValueOnce({ status: 200, data: MOCK_AMENDMENT_COVERENDDATE_CHANGE_USING_FACILITY_ENDDATE });
+      api.getFacility = jest.fn().mockResolvedValueOnce(gefFacility);
+
+      const req = {
+        params: {
+          _id: dealId,
+          amendmentId,
+          facilityId,
+        },
+        session,
+      };
+      await getAmendmentFacilityEndDate(req, res);
+
+      expect(res.render).toHaveBeenCalledWith('case/amendments/amendment-facility-end-date.njk', {
+        dealId,
+        facilityId,
+        dayInput: '',
+        monthInput: '',
+        yearInput: '',
+        isEditable: true,
+        user,
+      });
+    });
+
+    it('should render the template with isEditable false when amendment is found but has been completed', async () => {
+      const COMPLETED_AMENDMENT = { ...MOCK_AMENDMENT_COVERENDDATE_CHANGE_USING_FACILITY_ENDDATE, status: AMENDMENT_STATUS.COMPLETED };
+
+      api.getAmendmentById.mockResolvedValueOnce({ status: 200, data: COMPLETED_AMENDMENT });
+      api.getFacility = jest.fn().mockResolvedValueOnce(gefFacility);
+
+      const req = {
+        params: {
+          _id: dealId,
+          amendmentId,
+          facilityId,
+        },
+        session,
+      };
+      await getAmendmentFacilityEndDate(req, res);
+
+      expect(res.render).toHaveBeenCalledWith('case/amendments/amendment-facility-end-date.njk', {
+        dealId,
+        facilityId,
+        dayInput: '',
+        monthInput: '',
+        yearInput: '',
+        currentFacilityEndDate: undefined,
+        isEditable: false,
+        user,
+      });
+    });
+
+    it('should render template prepopulated with the previously entered field values if the facility end date has already been added to the amendment', async () => {
+      const previouslyEnteredFacilityEndDate = new Date(2025, 11, 11).toISOString();
```

Comment:
could the day, month, year be constants so they only need to be changed in one place